main/src/main/scala/sbt/plugins/SbtPlugin.scala

```diff
@@ -10,12 +10,19 @@ package sbt
 package plugins
 
 import sbt.Def.Setting
-import sbt.Keys._
+import sbt.Keys.*
+import sbt.SlashSyntax0.*
 
-object SbtPlugin extends AutoPlugin {
+object SbtPlugin extends AutoPlugin:
   override def requires = ScriptedPlugin
 
   override lazy val projectSettings: Seq[Setting[_]] = Seq(
-    sbtPlugin := true
+    sbtPlugin := true,
+    pluginCrossBuild / sbtVersion := {
+      scalaBinaryVersion.value match
+        case "3"    => sbtVersion.value
+        case "2.12" => "1.5.8"
```

Comment:
Why this version specifically?

Reply:
So the specific version doesn't really matter here but picking some old version allows the build user to be on a range of sbt versions. For some reason it's now 1.5 maybe because 1.2 is too old.